tests/countries/test_niger.py

```diff
@@ -0,0 +1,336 @@
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays import OPTIONAL
+from holidays.countries.niger import Niger, NE, NER
+from tests.common import CommonCountryTests
+
+
+class TestNiger(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(1960, 2050)
+        super().setUpClass(Niger, years=years, years_non_observed=years)
+        cls.no_estimated_holidays = Niger(years=years, islamic_show_estimated=False)
+        cls.optional_holidays = Niger(categories=OPTIONAL, years=years)
+
+    def test_country_aliases(self):
+        self.assertAliases(Niger, NE, NER)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(Niger(years=1958))
+
+    def test_new_years_day(self):
+        name = "Jour de l'An"
+        self.assertHolidayName(name, (f"{year}-01-01" for year in range(1998, 2050)))
+        obs_dt = (
+            "2012-01-02",
+            "2017-01-02",
+            "2023-01-02",
+        )
+        self.assertHolidayName(f"{name} (observé)", obs_dt)
+        self.assertNoNonObservedHoliday(obs_dt)
+        self.assertNoHolidayName(f"{name} (observé)", range(1960, 1998))
+
+    def test_easter_monday(self):
+        name = "Lundi de Pâques"
+        self.assertHolidayName(
+            name,
+            "2020-04-13",
+            "2021-04-05",
+            "2022-04-18",
+            "2023-04-10",
+            "2024-04-01",
+            "2025-04-21",
+        )
+        self.assertHolidayName(name, range(1960, 2050))
+
+    def test_concord_day(self):
+        name = "Fête nationale de la Concorde"
+        self.assertHolidayName(name, (f"{year}-04-24" for year in range(1995, 2050)))
+        self.assertNoHolidayName(name, range(1960, 1995))
+        obs_dt = (
+            "2011-04-25",
+            "2016-04-25",
+            "2022-04-25",
+        )
+        self.assertHolidayName(f"{name} (observé)", obs_dt)
+        self.assertNoNonObservedHolidayName(name, obs_dt)
+        self.assertNoHolidayName(f"{name} (observé)", range(1960, 1998))
+
+    def test_international_labor_day(self):
+        name = "Journée internationale du travail"
+        self.assertHolidayName(name, (f"{year}-05-01" for year in range(1960, 2050)))
+        obs_dt = (
+            "2011-05-02",
+            "2016-05-02",
+            "2022-05-02",
+        )
+        self.assertHolidayName(f"{name} (observé)", obs_dt)
+        self.assertNoNonObservedHolidayName(name, obs_dt)
+        self.assertNoHolidayName(f"{name} (observé)", range(1960, 1998))
+
+    def test_ascension_thursday(self):
+        name = "Ascension"
```

Comment:
```suggestion
        name = "Ascension"
        self.assertNoHolidayName(name)
```
Same for rest OPTIONAL holidays.